packages/run-protocol/src/reserve/collateralReserve.js

```diff
@@ -0,0 +1,218 @@
+// @ts-check
+
+import { E, Far } from '@endo/far';
+import { makeStore, keyEQ } from '@agoric/store';
+import { AmountMath } from '@agoric/ertp';
+import { CONTRACT_ELECTORATE, handleParamGovernance } from '@agoric/governance';
+import { offerTo } from '@agoric/zoe/src/contractSupport/index.js';
+
+import { AMM_INSTANCE_KEY, makeParamManager } from './params.js';
+
+const { details: X } = assert;
+
+const makeLiquidityKeyword = keyword => `${keyword}Liquidity`;
+
+/**
+ * Collateral Reserve holds onto Collateral for the RUN protocol, and can
+ * dispense it for various purposes under governance control. It currently
+ * supports governance decisions to add liquidity to an AMM pool.
+ *
+ * This contract has the ability to mint RUN. When adding liquidity to an AMM
+ * pool, it takes a specified amount of collateral on hand, and mints new RUN to
+ * accompany it, and deposits both into an AMM pool, using the AMM's method that
+ * allows the pool balance to be determined based on the contributed funds.
+ *
+ * @param {ZCF<
+ * {
+ *   electionManager: VoteOnParamChange,
+ *   main: {
+ *     AmmInstance: ParamRecord<'instance'>,
+ *     Electorate: ParamRecord<'invitation'>,
+ *   },
+ *   governedApis: ['addLiquidityToAmmPool'],
+ * }
+ * >} zcf
+ * @param {{feeMintAccess: FeeMintAccess, initialPoserInvitation: Payment}} privateArgs
+ */
+const start = async (zcf, privateArgs) => {
+  const {
+    main: {
+      [CONTRACT_ELECTORATE]: electorateParam,
+      [AMM_INSTANCE_KEY]: ammInstance,
+    },
+  } = zcf.getTerms();
+
+  /** @type {MapStore<Brand, Keyword>} */
+  const keywordForBrand = makeStore('keywords');
+  /** @type {MapStore<Keyword, [Brand, Brand]>} */
+  const brandForKeyword = makeStore('brands');
+
+  /**
+   *
+   * @param {*} issuer
+   * @param {string} keyword unique
+   * @returns {Promise<[argumentIssuer: Issuer, liquidityIssuer: Issuer]>}
+   */
+  const ammPublicFacet = E(zcf.getZoeService()).getPublicFacet(
```

Comment:
Looks like a stray doc comment.

```suggestion
  /** @type {XYKAMMPublicFacet} */
  const ammPublicFacet = E(zcf.getZoeService()).getPublicFacet(
```
And I think `XYKAMMPublicFacet` will need `makeAddLiquidityAtRateInvitation` added to it. We could define `XYKAMMPublicFacet` by the the return type of the contract `start`. E.g. in `multipoolMarketMaker.js`,
```
/** @typedef {ContractOf<start>['publicFacet']} XYKAMMPublicFacet */
```
but I don't know how to do that without losing the JSDoc on the current `XYKAMMPublicFacet` typedef.

Reply:
done. 

I we had had imputed types earlier, I might have added the doc on the individual methods. It doesn't seem worth distributing them now, though not having to keep the type declaration synced may make it worthwhile.